src/components/NavBar/OidcLoginComponent.jsx

```diff
@@ -15,53 +18,116 @@ import { ENV } from '../../constants';
  *
  * @memberof NavBar
  * @name OidcLoginComponent
+ * @param {object} Props - The props for OidcLoginComponent
+ * @param {React.Dispatch<React.SetStateAction<boolean>>} Props.setShowSignInModal
+ * - The set function for closing sign in modal
  * @returns {React.JSX.Element} - The OidcLoginComponent Component
  */
-const OidcLoginComponent = () => {
+const OidcLoginComponent = ({ setShowSignInModal }) => {
   const { login } = useSession();
-  const defaultOidc = ENV.VITE_SOLID_IDENTITY_PROVIDER || '';
+  const SUGGESTED_OIDC_OPTIONS = ENV.VITE_SUGGESTED_OIDC_OPTIONS?.split(', ') || [
+    'http://localhost:3000/'
+  ];
+  const defaultOidc = ENV.VITE_SOLID_IDENTITY_PROVIDER || SUGGESTED_OIDC_OPTIONS[0];
+  const [selectedOidcFromDropdown, setSelectedOidcFromDropdown] = useState(defaultOidc);
   const [oidcIssuer, setOidcIssuer] = useState(defaultOidc);
+  const [dropdownIsOpen, setDropdownIsOpen] = useState(false);
+
   const loginHandler = async () => {
     const redirectUrl = window.location.href;
+    localStorage.setItem('oidcIssuer', oidcIssuer);
     await login({ oidcIssuer, redirectUrl });
   };
+
+  const theme = useTheme();
+  const isSmallScreen = useMediaQuery(theme.breakpoints.down('sm'));
+
   return (
-    <>
-      <Box sx={{ flexGrow: 1 }} />
-      <TextField
-        type="text"
-        label="Pod Server URL"
-        variant="filled"
-        value={oidcIssuer}
-        onChange={(e) => setOidcIssuer(e.target.value)}
-        onKeyUp={(event) => {
-          if (event.key === 'Enter') {
-            loginHandler();
-            localStorage.setItem('oidcIssuer', oidcIssuer);
-          }
+    <Box
+      sx={{
+        display: 'flex',
+        flexDirection: isSmallScreen ? 'column' : 'row',
+        alignItems: 'center',
+        gap: '20px',
+        width: isSmallScreen ? '100%' : null
+      }}
+    >
+      <Autocomplete
+        id="pod-server-url"
+        sx={{
+          width: isSmallScreen ? null : '300px',
+          marginBottom: isSmallScreen && dropdownIsOpen ? '150px' : '0'
         }}
-        InputProps={{
-          disableUnderline: true
+        fullWidth={isSmallScreen}
+        options={SUGGESTED_OIDC_OPTIONS}
+        size="small"
+        freeSolo
+        includeInputInList
+        disablePortal
+        selectOnFocus
+        autoHighlight
+        blurOnSelect
+        openOnFocus
+        onOpen={() => setDropdownIsOpen(true)}
+        onClose={() => setDropdownIsOpen(false)}
+        value={selectedOidcFromDropdown}
+        inputValue={oidcIssuer}
+        onChange={(_, newValue) => {
```

Comment:
doesn't need to change but curious to know what the underscores are, and if they are necessary?

and differences between `onChange` and `onInputChange`.

Reply:
^ @rioredwards

Reply:
I know! MUI didn't make this super intuitive...

The value of the input can change when the user selects from the dropdown OR types:

`onChange` triggers when the user selects a value from the dropdown.

`onInputChange` triggers when the value of the input changes (i.e. the user selects a value from the dropdown OR types in the text box).

[Source](https://mui.com/material-ui/react-autocomplete/#controlled-states)

The underscore is a naming convention for variables that aren't actually used. In this case, the `onChange` callback receives two parameters: `(event, value)`, but here, I only care about `value`. Since I need to put something in that first position, I put an underscore to indicate it's not important. Syntax highlighters, ESlint, and other devs know about this convention and won't consider it an accidental unused variable.

[Source](https://stackoverflow.com/a/11406839/18654411)

Reply:
ok, is this `(_, value)` in lieu of `e.target.value`? Or are they different?